sdks/router-sdk/src/entities/trade.test.ts

```diff
@@ -210,6 +246,10 @@ describe('Trade', () => {
       const trade = await Trade.fromRoute(route, amount, TradeType.EXACT_INPUT)
       expect(trade.inputAmount.currency).toEqual(ETHER)
       expect(trade.outputAmount.currency).toEqual(token0)
+
+      expect(trade.amounts.inputAmountNative).toBeDefined()
```

Comment:
could assert that its 10?

Reply:
no because the inputAmountNative reflects the ETH that must be used for the route, and since this is a WETH input route its 0

Reply:
is the following true:
- if no weth input or output then inputAmount/outputAmount are undefined.
- if weth input, inputAmountNative is 0 and inputAmount is nonzero & defined
- if weth output, outputAmountNative is 0 and outputAmount is nonzero & defined
- if eth input, inputAmount nonzero & defined ? (only can be true on v4)
- if eth output, outputAmount nonzero & defined ? (only can be true on v4)
- if the route has both eth/weth in the route, then it must be entirely on v4 or a mixed route that includes a v4 path and inputAmount, outputAmount, inputAmountNative, and ouputAmountNative are all defined and non zero

Reply:
confirming that the following is true (and when we say input/output here I mean path input/output):

- if no weth input or output then inputAmountNative/outputAmountNative are undefined.
- if weth input, inputAmountNative is 0 and inputAmount is nonzero & defined (no change from legacy)
- if weth output, outputAmountNative is 0 and outputAmount is nonzero & defined (no change from legacy)
- if eth input, inputAmountNative nonzero & defined ? (only can be true on v4)
- if eth output, outputAmountNative nonzero & defined ? (only can be true on v4)
- if the route has both eth/weth in the route, then it must be entirely on v4 or a mixed route that includes a v4 path and inputAmount, outputAmount, inputAmountNative, and ouputAmountNative are all defined and non zero

Reply:
> if weth input, inputAmountNative is 0 and inputAmount is nonzero & defined (no change from legacy)

in this case if the true input (not path input) is WETH too, isnt `inputAmountNative` is undefined?

Reply:
yes undefined in that case